docs/reference/operators/btfgen.md

```diff
@@ -0,0 +1,16 @@
+---
+title: btfgen
```

Comment:
I'm not sure we want to be that explicit again. I wanted to explain layer vs data operators at the beginning to provide more context, but after that it gets complicated.
- wasm: Layer operator because handles a layer, but it's also a data operator because handles data
- bpf: pretty much the same, it also enriches some information 

I'd prefer to omit it by now. In the end, an operator being layer or data doesn't change the user experience.

Reply:
I decided to split it in two different folders. I'm still not sure was an operator can be both, but I don't have better ideas...

Reply:
I think the confusion of an operator being both layer and data operators comes from the wrong definition we were using for each type of operator:

- Layer: Handles layers and produce data but doesn't handle data (wrong because ebpf and wasm actually handle data).
- Data: Handles data but doesn't produce data (wrong because we want to produce data from some data op in the future).

Instead, to clarify things, I would suggest the following definition: Both type of operators **can** produce and handle data. The only difference is that layer operators does it by using additional data coming from a layer (and that's what makes them a layer operator):

- Wasm: Handle data using the WASM module present in the wasm layer.
- eBPF: Produce and handle ([formatters](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/operators/ebpf/formatters.go)) data using the eBPF program present in the ebpf layer.
- btfgen: Neither produce nor handle data (it's fine, we say they can, not they must). Instead, it uses files present in the bpfgen layer to allow Gadgets run in machines without BTF.

WDYT?

Reply:
I updated the documentation again not to have a strong differentiation of data and layer operators. I only mentioned that all operators can act over data and few of them handle layers of the OCI image. I think that's more clear.